docs/development-environment.md

```diff
@@ -1,298 +1,444 @@
+* ### Checking Jetpack's PHP for compatibility with different versions of PHP since 5.6
 
-### JETPACK__SANDBOX_DOMAIN
+	We have a handy `composer` script that will just run the PHP CodeSniffer `PHPCompatibilityWP` ruleset checking for code not compatible with PHP 5.6
+	
+	```sh
+	$ composer php:compatibility .
+	```
+	
+	_There's also a handy `yarn php:compatibility` that will run `composer php:compatibility` if you prefer_.
+	
+	```sh
+	$ yarn php:compatibility .
+	```
 
-External contributors do not need this constant.
-If you’re working on changes to the WordPress.com/server side of Jetpack, you’ll need to instruct your Jetpack installation to talk to your development server. Refer to internal documentation for detailed instructions.
+* ### Linting Jetpack's JavaScript
+	
+	`yarn lint` will check syntax and style in the following JavaScript pieces:
+	
+	* All the front end JavaScript that Jetpack relies on.
+	* All the JavaScript present in the Admin Page Single Page App for Jetpack.
+	
+	```sh
+	$ yarn lint
+	```
+	
+	_If you haven't done it yet, you may need to run `yarn` before `yarn lint` for installing node modules for this task_.
+
+---
+
+<a name="devtools"></a>
+
+# Standard development & debugging tools
+
+* ### WP_DEBUG
+	
+	You should do all Jetpack development with `define( 'WP_DEBUG', true );` in your `wp-config.php`, making sure that you’re not generating any Notices or other PHP issues in your error_log.
+
+* ### SCRIPT_DEBUG
+	
+	By default, WordPress loads minified versions of Jetpack's JS files. If you want to work with them, add `define( 'SCRIPT_DEBUG', true );` in your `wp-config.php`. This tells WordPress to load the non-minified JS version, allowing you to see your changes on page refresh. This applies to the JS files outside of `_inc/client/` and `extensions/`.
+
+* ### WP-CLI
+
+	Jetpack CLI is a command line interface for Jetpack, extending off of WP-CLI for WordPress. You can easily modify your installation of Jetpack with a just a few simple commands. All you need is SSH access and a basic understanding of command line tools.
+	
+	Usage:
+	
+	* `wp jetpack status [<full>]`
+	* `wp jetpack module <list|activate|deactivate|toggle> [<module_name>]`
+	* `wp jetpack options <list|get|delete|update> [<option_name>] [<option_value>]`
+	* `wp jetpack protect <whitelist> [<ip|ip_low-ip_high|list|clear>]`
+	* `wp jetpack reset <modules|options>`
+	* `wp jetpack disconnect <blog|user> [<user_identifier>]`
+	* `wp jetpack status`
+	* `wp jetpack status [<full>]`
+	
+	More info can be found in [our support documentation](https://jetpack.com/support/jetpack-cli/).
+
+* ### JETPACK_DEV_DEBUG
```

Comment:
Do we need those to be list items? Can't they just be sub-headings?

Reply:
The only reason I left this as a list is to indent them. It looks a bit neater to have them indented under the headline and the list seems to be the only way to achieve that.